### DISCUSSION TITLE
Inserting/escaping special characters in text

### DISCUSSION THREAD

**Discussion by arnaudgallou** [quarto-dev/quarto-cli#1182] [Q&A] [answered]
I'm having issues inserting special characters in text or tables when rendering as `pdf`. Special characters such as `≥` or greek letters don't show up unless I use a formula notation, which works or not depending on the context. E.g. `$\Delta$` works in text but throws an error when used in a table. Some unicode characters show up in table while others don't.

What is the recommended approach to insert or escape such characters?

``````
---
format: pdf
df-print: kable
---

4 $\geq$ 2

4$\geq$2

4 \≥ 2

$\Delta$MAT

\u0394MAT

\∆MAT

```{r}
#| echo: false
#| warning: false

library(tidyverse)
tibble(
  "$α$" = letters[1:2],
  "\u0394T" = 1:2,
  "\u03b2" = c("\u0394F~0-100~", "\u0394F~100-200~")
)
```
``````

Tested with Quarto v. 0.9.577 on RStudio v. 4.2.0, running on macOS Monterey.

**Comment by cderv** [comment, marked as answer]
Special character needs to be considered depending how you insert them.

When you are writing between `$ $`, you'll right them in an inline math environment where `\geq` is defined. If you want to add the mathematical symbol greater than or equal, it seems the right play. You could also write `$4 \geq 2$`. with `4$\geq$2` this is not identified as an equation, and `\geq` is not recognized. hence not shown in output. 

When you are writing in body, you are writing Markdown syntax. So if you want to use LaTeX directly to insert some symbol, you need to tell Pandoc to parse the content as raw LaTeX. For example, inserting Unicode character in LaTeX can work using XeLaTeX engine and special `\char` comment (Post about this https://agiletribe.wordpress.com/2015/04/07/adding-unicode-characters-to-latex-documents, and some doc here https://ctan.gutenberg-asso.fr/info/texbytopic/TeXbyTopic.pdf). `\symbol{"XXXX}` would work too with unicode code (https://tex.stackexchange.com/questions/298172/command-for-and-in-mathmode)

So inserted unicode by there hexadecimal would work using ``\symbol{"0394}`{=latex}MAT`. 

Regarding tables, your example work for me. So not sure why not for you, maybe a difference in version. I just know that using raw LaTeX in tables may require you to specificy `escape = FALSE` in `kable()`. Also when you provide `"\u0394T"` in the code chunk, this is processed by R, and the syntax used is a valid R syntax. 

``` r
cat("\u0394T")
#> ΔT
```

So there is unfortunately not one way to escape a special character as: 
- A character could be special in a context only. 
- The escaping to use depends on this context and where it is process. 

This 

```` 
---
format: pdf
keep-tex: true
df-print: kable
---

4 $\geq$ 2

$\Delta$MAT

`\symbol{"0394}`{=latex}MAT

```{r}
#| echo: false
#| warning: false

library(tidyverse)
tibble(
  "$α$" = letters[1:2],
  "\u0394T" = 1:2,
  "\u03b2" = c("\u0394F~0-100~", "\u0394F~100-200~")
)
```
````

give me this 

![image](https://user-images.githubusercontent.com/6791940/174584488-f8091061-51b1-43dc-b603-4dae125a9ec0.png)

Hope it helps

**Reply by arnaudgallou** [reply]
Thank you so much for the detailed reply. That's very helpful. I'm getting the same result as you. However the name of the third column in the table (which should be `β`) doesn't show up.

**Reply by cderv** [reply]
Interesting you are right. It shows correctly as beta in the `.tex` file (add `keep-tex: true` in YAML to see this intermediate file. 

However, to use greek symbol in tables, I usually do this 
````
tibble(
  "$\\alpha$" = letters[1:2],
  "$\\Delta$T" = 1:2,
  "$\\beta$" = c("\u0394F~0-100~", "\u0394F~100-200~")
) %>%
  knitr::kable(escape = FALSE)
````

**Comment by arnaudgallou** [comment]
I did a bit more testing to get my head around the different ways to insert special characters and noticed that table captions don't behave the same depending on the package used to produce the table. In the example below, Quarto interprets the caption of `tbl-knitr` properly and print 'β<sub>1</sub>'. However, for `tbl-kableExtra`, it shows `\beta1`. Also happens when using `gt`.

I know that `kableExtra` and `gt` do more stuff in the background than `knitr` but I thought the caption would be more or less independent. Is this a normal behavior? And most importantly, is there a fix for this? I actually need to print a `β` and have subscript digits in one of my table captions. Seems like unicode more or less works but for some reasons `β` still won't show. No clue for the subscript… `<sub></sub>`, `~` and `_` in formula don't work.

``````
---
format: pdf
execute: 
  echo: false
  warning: false
---

```{r}
library(magrittr)
```

```{r}
#| label: tbl-knitr
#| tbl-cap: "$\\beta$~1~"

tibble::tibble(x = rnorm(5)) %>% 
  knitr::kable()
```

```{r}
#| label: tbl-kableExtra
#| tbl-cap: "$\\beta$~1~"

tibble::tibble(x = rnorm(5)) %>% 
  kableExtra::kbl(booktabs = TRUE)
```
``````

**Reply by cderv** [reply]
yes I think you found a difference in the way table caption is treated. In first `kable()` will produce a Markdown table, in the second case, `kableExtra` will produce a LaTeX table. The caption is not treated the same way in the code base. Basically, in the second case, the math symbols are lost, and `\beta` is not parse in the math context. 

I'll open an issue about that. Thanks for the report!

**Reply by cderv** [reply]
See https://github.com/quarto-dev/quarto-cli/issues/1185
